Sources/Afluent/SequenceOperators/ThrottleSequence.swift

```diff
@@ -0,0 +1,165 @@
+            let intervalEvents = IntervalEvents()
+            
+            public mutating func next() async throws -> Element? {
+                try Task.checkCancellation()
+                
+                let clock = self.clock
+                let interval = self.interval
+                let upstream = self.upstream
+                let intervalEvents = self.intervalEvents
+                
+                if iterator == nil {
+                    self.iterator = AsyncThrowingStream<(Element?, Element?), Error> { continuation in
+                        
+                        let intervalTask = DeferredTask {
+                            if let intervalStartInstant = intervalEvents.startInstant {
+                                intervalEvents.updateHasStartedInterval(true)
+                                
+                                let intervalEndInstant = intervalStartInstant.advanced(by: interval)
+                                try await clock.sleep(until: intervalEndInstant, tolerance: nil)
+                                
+                                let firstElement = intervalEvents.firstElement
+                                let latestElement = intervalEvents.latestElement
+                                
+                                continuation.yield((firstElement, latestElement))
+                                
+                                intervalEvents.updateFirst(element: nil)
+                                intervalEvents.updateHasStartedInterval(false)
+                            }
+                        }
+                        
+                        Task {
+                            do {
+                                for try await el in upstream {
+                                    if !intervalEvents.hasSeenFirstElement {
+                                        continuation.yield((el, el))
+                                        intervalEvents.updateHasSeenFirstElement()
+                                        continue
+                                    }
+                                    if intervalEvents.firstElement == nil {
+                                        intervalEvents.updateStart(instant: clock.now)
+                                        intervalEvents.updateFirst(element: el)
+                                        intervalTask.run()
+                                    }
+                                    intervalEvents.updateLatest(element: el)
+                                }
+                                if intervalEvents.hasStartedInterval {
+                                    continuation.yield((intervalEvents.firstElement, intervalEvents.latestElement))
+                                }
+                                continuation.finish()
+                            } catch {
+                                continuation.yield((intervalEvents.firstElement, intervalEvents.latestElement))
```

Comment:
Maybe I don't understand something, but I'm a bit confused why we're yielding events on an error? Wouldn't this potentially break the throttle contract? (E.g. I receive an error after < the internal time, we'd publish before the internal). Shouldn't we just immediately finish with error?

Reply:
Discussed on Slack. This actually is the same behavior as the Combine operator. My only ask is that we add a test for the case when an error is thrown before the throttle interval ends.